adb/common.py

```diff
@@ -190,10 +193,14 @@ def BulkRead(self, length, timeout_ms=None):
           'Could not receive data from %s (timeout %sms)' % (
               self.usb_info, self.Timeout(timeout_ms)), e)
 
+  def BulkReadAsync(self, length, timeout_ms=None):
+    # See: https://pypi.python.org/pypi/libusb1 "Asynchronous I/O" section
+    return
+
   @classmethod
   def PortPathMatcher(cls, port_path):
     """Returns a device matcher for the given port path."""
-    if isinstance(port_path, basestring):
+    if isinstance(port_path, str):
```

Comment:
basestring was removed in Python3. Using str here instead for py2/3 compatibility.